### PR TITLE
GlossaryPage and SearchResults to functional components

### LINKED ISSUE #121: Atualizar componentes de classe para fazerem uso de hooks
**Descrição da melhoria:**
Seria legal nossos componentes terem um formato mais funcional, tornando-os mais intuitivos e para ajudar com isso, o React provê as [`hooks`](https://pt-br.reactjs.org/docs/hooks-intro.html)

### PR DESCRIPTION
**Pull request by JKarlavige**
Fixes #121

Replaces GlossaryPage and SearchResults Class Components with stateless functional components. This allows React Hooks to be implemented if state management is needed.

### CODE REVIEW

File: src/components/glossary/GlossaryPage.js [hunk 1]
```diff
@@ -1,63 +1,65 @@
...
+import SearchResults from './results/SearchResults';
 
 const DayPhrase = ({ entry }) => {
-    return (
-        <span className={"glossary__day-phrase"}>
-            Você sabe o que é <Link className="emphasis pointer-hover light-accent lighter-hover" to={`/${entry}`}>{ entry }</Link>?
-        </span>)
+  return (
+    <span className={'glossary__day-phrase'}>
+      Você sabe o que é{' '}
```
Comment (fanny) [inline src/components/glossary/GlossaryPage.js]: Current, this `{' '}` has some finality?

File: src/components/glossary/GlossaryPage.js [hunk 2]
```diff
@@ -1,63 +1,65 @@
...
+    return entries[index];
+  };
+
+  const getTerm = () => props.match.params.term;
+
+  const isSearchEmpty = () => {
+    return getTerm() === undefined;
+  };
+
+  const randomEntry = getRandomEntry();
```
Comment (fanny) [inline src/components/glossary/GlossaryPage.js]: you can remove this variable, and invoke directly the method, when necessary

File: src/components/glossary/results/SearchResults.js [hunk 3]
```diff
@@ -1,36 +1,31 @@
-import React, {Component} from "react";
-import "./SearchResults.css"
+import React, { Component } from 'react';
```
Comment (fanny) [inline src/components/glossary/results/SearchResults.js]: Remove the component import
```suggestion
import React from 'react';
```

File: src/components/glossary/results/SearchResults.js [hunk 4]
```diff
@@ -1,36 +1,31 @@
...
-            </div>
-        );
-    }
-}
+import { TermCard } from '../../common/index';
+import NotFound from './404/NotFound';
+
+const Results = ({ results }) =>
+  results.map(result => (
+    <TermCard term={result} key={result.entry + result.meaning} />
```
Comment (fanny) [inline src/components/glossary/results/SearchResults.js]:
```suggestion
   <TermCard term={result} key=`${result.entry}${result.meaning}` />
```

**Review by fanny — CHANGES_REQUESTED** [review]
Thanks for this contribution! @JKarlavige  :heart_decoration:  Only some nitpicks

**Review by pedroespindula — DISMISSED** [review]
Hello @JKarlavige, thanks to you for contributing to OpenDevUFCG. We're very happy to see people interested on what we are doing. I saw your PR and it's all right, you made interesting and useful refactors to our components. However you commited a package-lock.json file. We already have a lock file, but it's from yarn. I don't know if we should keep both of then. I searched the pros and cons of having both and i still don't know the answer for this question. Soo, for me your PR it's approved but i still want that someone else review you.  Thanks again, be welcome to contribute again!

File: src/components/glossary/GlossaryPage.js (hunk 1 above)
Comment (JKarlavige) [reply src/components/glossary/GlossaryPage.js]: Hi @fanny, I appreciate the suggestions! They're super helpful in learning a bit more about ideal ways to structure code in React :)

Comment (fanny) [reply src/components/glossary/GlossaryPage.js]: You're welcome!

**Pull request merged**